MAINT: Remove nickname from polynomial classes.

rossbar opened the pull request:
This is a bit of proposed cleanup following the changes to polynomial printing in #15666.

The convenience classes derived from ABCPolyBase have a `nickname` attribute that was only used internally in the previous implementation of `__str__`. After the overhaul of `__str__` in #15666, this attr is no longer used anywhere internally, nor is it documented.

This change would technically break backward compatibility as the attribute was not prepended with `_`, but as it was an undocumented attribute that was only used internally in one specific place (and is no longer used anywhere) I thought it might be worth at least considering its removal.

rossbar committed:
MAINT: Remove nickname from polynomial classes.

The convenience classes derived from ABCPolyBase had a nickname
attribute that was only used internally in the previous
implementation of __str__. After the overhaul of __str__ in #15666,
this attr is no longer used.

rossbar commented:
re-trigger CI

seberg commented:
@rossbar I think we should go ahead with it unless anyone speaks up. Its undocument, so the only one relying on it would be someone who creates new polynomials and even then it is very unlikely to affect them (worst case they have to replace the `__repr__`). Maybe write a sentence in the release notes in any case?

rossbar committed:
DOC: Add release note.

Add release note to notify users of removal of the abstract
property, and highlight users that may be affected by the
change.

rossbar commented:
Thanks for the feedback @seberg! I added a release note that summarizes the change and highlights users whom the change may affect.

rossbar committed:
DOC: fixed rST in release note

seberg commented:
Thanks Ross, lets put it in now, if anyone complains before the next release we can still undo.
